grammar/jbuild

```diff
@@ -0,0 +1,42 @@
+(jbuild_version 1)
+
+(library
+ ((name grammar)
+  (public_name coq.grammar)
+  (wrapped false)
+  (flags (-rectypes -w -58))
+  (libraries (camlp5))))
+
+; Custom camlp5! This is a net speedup, and a preparation for using
+; Dune's preprocessor abilities.
+(rule
+ ((targets (coqmlp5))
+  (action (run mkcamlp5.opt pa_o.cmx pa_op.cmx pr_dump.cmx pa_extend.cmx q_MLast.cmx pa_macro.cmx pr_o.cmx -o coqmlp5))))
+
+(rule
+ ((targets (coqp5))
+  (action (run mkcamlp5.opt pa_o.cmx pa_op.cmx pr_dump.cmx pa_extend.cmx q_MLast.cmx pa_macro.cmx pr_o.cmx ${path:grammar.cmxa} -o coqp5))))
+
+(install
+ ((section bin)
+ (files (coqp5 coqmlp5))))
+
+(rule
```

Comment:
I don't know dune, so maybe the answer is sadly no, but I see a lot of copy/paste here.  Can it be eliminated?

Reply:
We could eliminate it now by generating the rules [as we do for plugins] but there are not so many rules, I think 7/8 in total so I didn't bother.

In the future Dune will indeed support a more expressive rule language and this cut and paste will go away.